src/store/ViewStore.ts

```diff
@@ -135,7 +135,7 @@ export default class ViewStore {
         }
         this.mainStore.state.setChartIsReady(false);
         const stx = this.stx;
-        const granularity = getIntervalInSeconds(ViewStore.views[idx].layout) as TGranularity;
+        const granularity = getIntervalInSeconds(this.sortedItems[idx].layout) as TGranularity;
```

Comment:
- You can define a variable for sortedItems and use it multiple places instead of calling the sort function multiple times.
- Why are you passing `idx` to the function?

Reply:
Alrighty will create a variable. 
For the idx, so that the system will know which template im clicking. if not it will return all the idx